app/javascript/components/student/Editor.tsx

```diff
@@ -111,29 +217,84 @@ export function Editor({
             return
           }
 
-          dispatch({ type: EditorStatus.SUBMISSION_CANCELLED })
+          dispatch({ type: ActionType.SUBMISSION_CANCELLED })
         })
         .finally(() => {
           controllerRef.current = undefined
         })
     },
     [abort, controllerRef]
   )
+  const submit = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      if (!submission) {
+        return
+      }
+
+      dispatch({ type: ActionType.CREATING_ITERATION })
+
+      fetchJSON(submission.links.submit, { method: 'POST' }).then(
+        (json: any) => {
+          dispatch({ type: ActionType.ITERATION_CREATED })
+        }
+      )
+    },
+    [submission]
+  )
   const cancel = useCallback(() => {
     abort()
-    dispatch({ type: EditorStatus.SUBMISSION_CANCELLED })
+    dispatch({ type: ActionType.SUBMISSION_CANCELLED })
   }, [dispatch, abort])
+  const editorDidMount = useCallback(
+    (editor) => {
+      editorRef.current = editor
+    },
+    [editorRef]
+  )
+  const updateSubmission = useCallback(
+    (submission) => {
+      dispatch({
+        type: ActionType.SUBMISSION_CHANGED,
+        payload: { submission: submission },
+      })
+    },
+    [dispatch]
+  )
 
   useEffect(() => {
     return abort
   }, [abort])
 
   return (
     <div>
-      <CodeEditor onSubmit={submit} />
-      {status === EditorStatus.SUBMITTING && <Submitting onCancel={cancel} />}
+      <MonacoEditor
+        width="800"
+        height="600"
+        language="ruby"
+        editorDidMount={editorDidMount}
+        defaultValue="Code"
+      />
+      <button type="button" onClick={runTests}>
+        Run tests
+      </button>
+      <button
+        type="button"
+        onClick={submit}
+        disabled={submission && submission.testsStatus !== TestRunStatus.PASS}
+      >
+        Submit
+      </button>
+      {status === EditorStatus.CREATING_SUBMISSION && (
```

Comment:
I decided to just lump all of these status messages here. Hopefully this will make it easier for us to move it around once we start styling.